### DISCUSSION TITLE
How can I add user to a group?

### DISCUSSION THREAD

**Discussion by TacticalPenGuinn** [lima-vm/lima#691] [Q&A] [answered]
Hello,

Is there any functionality in my image's yaml.cfg  or elsewhere to add the running instant's user to a specific group or custom made group?

**Comment by jandubois** [comment, marked as answer]
No, there isn't. You have to do it manually via `limactl shell`, or include the commands in a provisioning script.

**Reply by lizrice** [reply]
Is there a variable for the user name that could be used in a provisioning script?

**Reply by msgilligan** [reply]
Did anyone figure out how to do this (get the username in a variable for adding the user to a group) in a provisioning script?

**Reply by afbjorklund** [reply]
The username is available in `$LIMA_CIDATA_USER`, but the primary group currently doesn't have a variable.

**Reply by afbjorklund** [reply]
Note that extra groups don't take effect immediately, so you have to use `newgrp` or some other workaround.

**Reply by msgilligan** [reply]
FYI: I'm trying to configure Podman in a Debian Bookworm instance and need need to add the user to the `systemd-journal` group.

**Reply by msgilligan** [reply]
> so you have to use `newgrp` or some other workaround.

Is there a workaround that works with Lima provisioning so that when the VM finishes booting the user had been added to the group?  (`newgrp` doesn't do that because it starts a sub shell)

**Reply by jandubois** [reply]
I have used `reboot` in a provisioning script to accomplish a clean setup. You just need to make sure you don't end up in an endless reboot loop. But otherwise Lima deals with rebooting while waiting for requirements just fine. It bothered me at first, but since it only reboots once during the first boot, it doesn't really matter.

**Reply by msgilligan** [reply]
I was wondering if I could put it in the `mode: boot` section of provisioning and create the group before the user shell is started...

**Reply by jandubois** [reply]
This might work; I haven't tried it. But I just noticed the `mode: boot` is not implemented in the `lima-init` script used in our default Alpine image, so it won't work there for now.

**Reply by jandubois** [reply]
Was just about to create an issue for this, but then saw that I already did this some time ago: https://github.com/lima-vm/alpine-lima/issues/87

**Reply by jandubois** [reply]
@msgilligan Please report back here if you try the `boot` script, and it actually works. On further thoughts I think it wouldn't work on Alpine/OpenRC even if we supported `bootcmds` because they would just run from `lima-init-local` instead of `lima-init`. I don't know if systemd would behave differently, but would like to know if it does works contrary to my expectation!

**Reply by msgilligan** [reply]
I tried:
```
- mode: boot
  script: |
    #!/bin/bash
    set -eux
    usermod -aG systemd-journal $LIMA_CIDATA_USER
```

This is the result:
```
/run/cloud-init/tmp/tmprbrvm5x4.sh: 4: LIMA_CIDATA_USER: parameter not set
2023-07-26 18:15:03,094 - util.py[WARNING]: Failed to run bootcmd module bootcmd
2023-07-26 18:15:03,095 - util.py[WARNING]: Running module bootcmd (<module 'cloudinit.config.cc_bootcmd' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_bootcmd.py'>) failed
```

I also had to remove the `-o pipefail` option from the `set -eux -o pipefail` line that I had copied to g

**Reply by jandubois** [reply]
Thanks, I've turned your comment into a new issue at #1685

**Reply by msgilligan** [reply]
and if I hard-code the username it fails on the very first run with:
```
usermod: user 'sean' does not exist
```

**Reply by msgilligan** [reply]
I'm beginning to think it might be nice to have an `additionalGroups` setting in the `.yaml` that adds the user to those groups upon creation.

**Reply by jandubois** [reply]
Please create an enhancement request issue for this!

**Reply by msgilligan** [reply]
https://github.com/lima-vm/lima/issues/1686 created.
